Assets/BossRoom/Scripts/Infrastructure/NetworkStats.cs

```diff
@@ -124,6 +130,27 @@ void FixedUpdate()
                 if (m_TextStat != null)
                 {
                     m_TextToDisplay = $"RTT: {(m_BossRoomRTT.Average * 1000).ToString("0")} ms;\nUTP RTT {m_UtpRTT.Average.ToString("0")} ms";
+                    if (m_UtpRTT.Average > k_BadNetworkConditionsRTTThreshold)
+                    {
+                        m_TextStat.color = Color.red;
+                    }
+                    else if (m_UtpRTT.Average > k_StrugglingNetworkConditionsRTTThreshold)
+                    {
+                        m_TextStat.color = Color.yellow;
+                    }
+                    else
+                    {
+                        m_TextStat.color = Color.white;
+                    }
+                }
+
+                if (m_TextBadNetworkConditions != null)
+                {
+                    // Right now, we only base this warning on UTP's RTT metric, but in the future we could watch for packet loss as well, or other metrics.
```

Comment:
```suggestion
                    // Right now, we only base this warning on UTP's RTT metric, but in the future we could watch for packet loss as well, or other metrics.
                    // This could be a simple icon instead of doing heavy string manipulations.
```